packages/run-protocol/src/runStake/attestation.js

```diff
@@ -0,0 +1,256 @@
+    );
+
+    // If setLiened returns successfully, then
+    // the preconditions were indeed met and we
+    // can update the stored liened amount.
+    await E(lienBridge).setLiened(address, current, target);
+    amountByAddress.set(address, target);
+    return attestationPaymentP;
+  };
+
+  /** @param {ZCFSeat} seat */
+  const returnAttestation = async seat => {
+    assertProposalShape(seat, { give: { Attestation: null }, want: {} });
+
+    /** @type {Amount<'copyBag'>} */
+    const attestationAmount = seat.getAmountAllocated(KW.Attestation);
+    const { address, lienedAmount: amountReturned } =
+      unwrapLienedAmount(attestationAmount);
+
+    const liened = amountByAddress.get(address); // or throw
+    const updated = subtract(liened, amountReturned); // or throw
+
+    await E(lienBridge).setLiened(address, liened, updated);
+    // COMMIT. Burn the escrowed attestation payment and exit the seat
+    amountByAddress.set(address, updated);
+    zcfMint.burnLosses(harden({ Attestation: attestationAmount }), seat);
+    seat.exit();
+  };
+
+  /**
+   * Get liened amount synchronously.
+   *
+   * @param {string} address
+   * @param {Brand<'nat'>} brand
+   */
+  const getLiened = (address, brand) => {
+    assert.typeof(address, 'string');
+    assert.equal(
+      brand,
+      stakeBrand,
+      X`This contract can only make attestations for ${brand}`,
+    );
+    return getOrElse(amountByAddress, address, () => empty);
+  };
+
+  const lienMint = harden({
+    mintAttestation,
+    returnAttestation,
+    getLiened,
+    /** @param { Amount<'copyBag'> } attAmt */
+    unwrapLienedAmount: attAmt => unwrapLienedAmount(attAmt).lienedAmount,
+  });
+  return harden({ issuer, brand: attBrand, lienMint });
+};
+
+/**
+ * Authorize each account holder to lien some of their staked assets
+ * and get an attestation that the lien is in place.
+ *
+ * @param {ZCF} zcf
+ * @param {Brand} stakeBrand
+ * @param {string} keyword for attestation issuer
+ * @param {ERef<StakingAuthority>} lienBridge
+ *
+ * NOTE: the liened amount is kept both here in JS and on the
+ * other side of the lienBridge. The mint created here
+ * is the authority on the liened amount. The liened amount
+ * on the other side of the bridge may lag. Do not create two
+ * mints that share a bridge.
+ *
+ * The provideAttestationMaker method on the returned creatorFacet
+ * provides the authorization for an address. The account holder
+ * can then use attMaker.makeAttestation(lienDelta).
+ *
+ * Note the Attestation keyword for use in returnAttestation offers.
+ */
+export const makeAttestationFacets = async (
+  zcf,
+  stakeBrand,
+  keyword,
+  lienBridge,
+) => {
+  /** @type {Store<Address, AttMaker>} */
+  const attMakerByAddress = makeStore('address');
+
+  const { issuer, brand, lienMint } = await makeAttestationKit(
+    zcf,
+    keyword,
+    stakeBrand,
+    lienBridge,
+  );
+
+  /** @param { string } address */
+  const makeAttMaker = address =>
```

Comment:
I assume "Att" stands for "attestation" but the returned object is more than an attestation maker. is this an "attestation kit"?

Reply:
The `makeXKit` convention is for making an `X` plus some related objects. Here we return exactly one object that has a `makeAttestation` method; that is: we return an attestation maker.

Reply:
> The `makeXKit` convention is for making an `X`

Ah, there's always an X object in the kit, got it. 

> plus some related objects

Just want to confirm that Kit can includes related _functions_ as well. E.g. `makePromiseKit` returns `{ promise, reject, resolve }`.

> Here we return exactly one object that has a `makeAttestation` method

I understand that if it were `makeAttestionKit` then it would have to be `{ attestion, …relatedStuff }`

> that is: we return an attestation maker.

Isn't `makeAttestation` function the attestation maker? The object being returned here has `makeAttestion` and several related functions. Maybe `Kit` isn't the right term but there's something wrapping the actual attestation maker. 

By my read, this far object is the "attestation tools" and the current `makeAttestationTools` function is better understood as `makeAttestationFacets`. Though I like your idea below to return wrappers, in which case this would be `makeAttestionWrappers`.

Though I'd go further and adopt the pattern from https://github.com/Agoric/agoric-sdk/blob/master/packages/governance/src/contractHelper.js#L100-L106 and name this something like `handleAttestations`.

Reply:
This is another decision that I inherited and so didn't feel obliged to do any better. But it's not standing up under scrutiny. So I'll reconsider.